javascriptv3/example_code/ssm/tests/create-maintenance-window.unit.test.js

```diff
@@ -0,0 +1,92 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { describe, it, expect, vi, beforeEach } from "vitest";
+import { CreateMaintenanceWindowCommand, SSMClient } from "@aws-sdk/client-ssm";
+import { main } from "../actions/create-maintenance-window.js";
+
+describe("createMaintenanceWindow", () => {
+  beforeEach(() => {
+    vi.resetAllMocks();
+  });
+
+  it("should create a new maintenance window successfully", async () => {
+    const mockWindowId = "test-window-id";
+    const sendMock = vi
+      .spyOn(SSMClient.prototype, "send")
+      .mockResolvedValueOnce({
+        windowId: mockWindowId,
+      });
+
+    const result = await main({
+      name: "test-window",
+      description: "Test maintenance window",
+      allowUnassociatedTargets: true,
+      duration: 2,
+      cutoff: 1,
+      schedule: "cron(0 0 ? * MON *)",
+    });
+
+    expect(sendMock).toHaveBeenCalledWith(
+      expect.any(CreateMaintenanceWindowCommand)
+    );
+    expect(sendMock).toHaveBeenCalledWith(
+      expect.objectContaining({
+        input: {
+          Name: "test-window",
+          Description: "Test maintenance window",
+          AllowUnassociatedTargets: true,
+          Duration: 2,
+          Cutoff: 1,
+          Schedule: "cron(0 0 ? * MON *)",
+        },
+      })
+    );
+    expect(result.WindowId).toEqual(mockWindowId);
```

Comment:
Ah, this is why you're returning those objects. That's fine, but the actions should definitely be doing more than just returning these.

Reply:
Added logs before the returns.